diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt

```diff
@@ -0,0 +1,294 @@
+import org.cqfn.diktat.ruleset.utils.*
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import org.jetbrains.kotlin.com.intellij.psi.tree.IElementType
+
+
+/**
+ * This class handles rule 2.6
+ * Part 1:
+ * there must be 1 space between the comment character and the content of the comment;
+ * there must be a newline between a Kdoc and the previous code above and no blank lines after.
+ * No need to add a blank line before a first comment in this particular name space (code block), for example between function declaration and first comment in a function body.
+ *
+ * Part 2:
+ * Leave one single space between the comment on the right side of the code and the code.
+ * Comments in if else should be inside code blocks. Exception: General if comment
+ */
+class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdoc-comments-codeblocks-formatting") {
+
+    companion object {
+        private const val MAX_SPACES = 1
+        private const val APPROPRIATE_COMMENT_SPACES = 1
+    }
+
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        isFixMode = autoCorrect
+        emitWarn = emit
+
+        val configuration = CommentsFormattingConfiguration(
+                configRules.getRuleConfig(COMMENT_WHITE_SPACE)?.configuration ?: mapOf())
+
+
+        when (node.elementType) {
+            CLASS, FUN, PROPERTY -> {
+                checkBlankLineAfterKdoc(node, EOL_COMMENT)
+                checkBlankLineAfterKdoc(node, KDOC)
+                checkBlankLineAfterKdoc(node, BLOCK_COMMENT)
+            }
+            IF -> {
+                handleIfElse(node)
+            }
+            EOL_COMMENT, BLOCK_COMMENT -> {
+                handleEolAndBlockComments(node, configuration)
+            }
+            KDOC -> {
+                handleKdocComments(node, configuration)
+            }
+        }
+    }
+
+    private fun checkBlankLineAfterKdoc(node: ASTNode, type: IElementType) {
+        val kdoc = node.getFirstChildWithType(type)
+        val nodeAfterKdoc = kdoc?.treeNext
+        if (nodeAfterKdoc?.elementType == ElementType.WHITE_SPACE && nodeAfterKdoc.numNewLines() > 1) {
+            WRONG_NEWLINES_AROUND_KDOC.warnAndFix(configRules, emitWarn, isFixMode, kdoc.text, nodeAfterKdoc.startOffset, nodeAfterKdoc) {
+                nodeAfterKdoc.leaveOnlyOneNewLine()
+            }
+        }
+    }
+
+    private fun handleKdocComments(node: ASTNode, configuration: CommentsFormattingConfiguration) {
+        if (node.treeParent.treeParent.elementType == BLOCK) {
```

Comment:
What if KDoc is located on top level, for example before package directive? Will `treeParent.treeParent` still be valid?

Reply:
Actually, if you check [CI run of snapshot](https://github.com/cqfn/diKTat/pull/235/checks?check_run_id=1121890916) you can see there is an error somewhere in your rule